pyOCD/utility/conversion.py

```diff
@@ -37,6 +38,22 @@ def word2byte(data):
         res.append((x >> 24) & 0xff)
     return res
 
+## @brief Convert a byte array into a halfword array.
+def byte2half(data):
+    byteData = []
+    for i in data:
+        byteData.extend([i & 0xff, (i >> 8) & 0xff])
```

Comment:
This should be data[i] getting added to byteData, rather than just 'i'.  Also this function should swapped with half2byte.